xdsl/dialects/x86/ops.py

```diff
@@ -314,18 +315,43 @@ class RR_XorOp(R_RR_Operation[GeneralRegisterType, GeneralRegisterType]):
 
 
 @irdl_op_definition
-class RR_MovOp(R_RR_Operation[GeneralRegisterType, GeneralRegisterType]):
+class RR_MovOp(X86Instruction, X86CustomFormatOperation):
```

Comment:
I would recommend against this change. The reason for the helpers is to make the code for each instruction a bit more readable. The base class naming convention in this file is:

`R` * number of destination registers `_` `R` * number of source registers `_` I * number of immmediate values `_` `Operation`

So this line should read:
```suggestion
class R_R_MovOp(R_R_Operation[GeneralRegisterType, GeneralRegisterType):
```

I can't see in the GH UI whether this exists already, could you please add it if not? Similarly for the move immediate operation below.

Reply:
As noted in the commit message, I found no suitable base classes for these instructions. I do agree, though, that factoring out the boilerplate would make things more readable.

`R_R_Operation` exists, but it’s “a base class for x86 operations that have one register acting as both source and destination”. But here we need two registers, one destination and one source. Do you think there’s a way to disambiguate these two cases within the current naming scheme?

Reply:
I think that's just bad phrasing, it's defined as having one operand and one result, so the doc comment should be changed, but it seems like the definition is good.

Reply:
The `R_R_Operation` is implemented as described in the class docstring, as can be seen here: https://github.com/xdslproject/xdsl/blob/841db977a0d25e9a4ec6a09b9311ce099a9bfdcf/xdsl/dialects/x86/ops.py#L431

Reply:
I see, it looks like this scheme is not quite pulling its weight then, and that these definitions are missing an explicit constraint that enforces that the registers are the same. I would probably rename the existing op `R_R_Inout_Operation`, and add an `R_R_Operation` for the case where the registers aren't the same.